packages/metro-resolver/src/__tests__/package-exports-test.js

```diff
@@ -211,6 +211,40 @@ describe('with package exports resolution enabled', () => {
       `);
     });
 
+    test('[nonstrict] should fall back to file resolution and log warning when subpath is not listed', () => {
```

Comment:
nit: Ideally, can we split this new test into a separate PR? Or convince me why it belongs here :)

Reply:
hmm this is a test that I started implementation with. It was checking for a behaviour that was not spec compliant. I'm not sure if it makes sense to keep it at all now.

I've removed it from this PR and can add it inside of a separate PR if you want, up to you :)